### ISSUE TITLE
Exception in thread "main" java.nio.file.NoSuchFileException: BabelfishCompassUser.Optimistic.cfg

### ISSUE THREAD

**Issue by adnan79git** [babelfish-for-postgresql/babelfish_compass#102]
Unable to run BabelFish due to the following error:-
```
C:\BabelfishCompass>BabelfishCompass.bat MyReport hhscript.sql
Babelfish Compass v.2023-03, March 2023
Compatibility assessment tool for Babelfish for PostgreSQL
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

Reading BabelfishFeatures.cfg
Latest Babelfish version supported: 3.1.0: BabelfishFeatures.cfg
Exception in thread "main" java.nio.file.NoSuchFileException: BabelfishCompassUser.Optimistic.cfg -> C:\Users\xxxx\Documents\BabelfishCompass\BabelfishCompassUser.Optimistic.cfg
        at sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
        at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
        at sun.nio.fs.WindowsFileCopy.copy(Unknown Source)
        at sun.nio.fs.WindowsFileSystemProvider.copy(Unknown Source)
        at java.nio.file.Files.copy(Unknown Source)
        at compass.CompassUtilities.installOptimisticCfgFile(CompassUtilities.java:6965)
        at compass.CompassConfig.validateCfgFile(CompassConfig.java:1624)
        at compass.Compass.main(Compass.java:843)
C:\BabelfishCompass>
```

**Comment by rcv-aws** [comment]
That's weird. Could you please:
- unzip the Compass .zip file into a new folder and do a dir of its contents, and post that here?
- check if you have a file named **C:\Users\xxxx\Documents\BabelfishCompass\BabelfishCompassUser.Optimistic.cfg**  (there is a folder named C:\Users\xxxx\Documents\BabelfishCompass, correct?)

**Comment by rcv-aws** [comment]
Can you please replace your compass.jar file with the one attached here, and retry?
[compass.zip](https://github.com/babelfish-for-postgresql/babelfish_compass/files/11020844/compass.zip)

**Comment by ITfromHell** [comment]
Hi,
I had same problem, testet the compass.jar from your ZIP and it works!
Testet at OSX Catalina 10.15.7 and Windows Server 2019 1809 with JRE 1.8.0_361

Now I´m curios to the results of the tool ;-)
Thanks a lot!

**Comment by rcv-aws** [comment]
Okay ... but I'd still like to understand what happened. 'cos I am unable to reproduce the scenario which matches your stack trace. The new .jar just doesn't exist when that condition occurs.
Would you mind checking the following:
- does your Babelfish installation directory contain a file named **BabelfishCompassUser.Optimistic.cfg.2023-Mar-20-HH.MM.SS**  ?
- do you have a file named **C:\Users\<yourname>\Documents\BabelfishCompass\BabelfishCompassUser.Optimistic.cfg**
- are the contents of those two files identical?

I'd really like to solve this so I'd greatly appreciate if you could shine a bit more light on it. Thanks in advance!

**Comment by adnan79git** [comment]
The contents of my Babelfish installation directory are as follows:-
![babelfish-installation](https://user-images.githubusercontent.com/78516165/226467495-a22b8ab7-db04-4349-a6cb-8cc1adb326f2.PNG)

The contents of C:\Users<yourname>\Documents\BabelfishCompass directory:- 
![babelfish-doc](https://user-images.githubusercontent.com/78516165/226467785-8b60f84a-2531-40c2-8b6d-8de0c6eed1b9.PNG)

As we can see the contents are not identical. I think I downloaded the installation zip file and just unzipped in a folder and then started to use it on the command line. No other installation step was taken.

**Comment by rcv-aws** [comment]
Thanks. 
When you run Compass, the first thing it does is copy **BabelfishCompassUser.Optimistic.cfg** from the installation directory to **C:\Users<yourname>\Documents\BabelfishCompass**. 
This is the step that originally failed in your first post. It looks like it still fails, and Compass prints a message about it, but it doesn't stop you from proceeding. 
Would you mind checking your console output directly following the line 
`Latest Babelfish version supported: 3.1.0: BabelfishFeatures.cfg`

Thanks!

**Comment by adnan79git** [comment]
I deleted previously downloaded folder and did a fresh download and unzipped. Ran powershell as admin - and gave another go at generating the report. This time around it went without a glitch. Probably needed admin privileges or this latest download addresses the issue.
I can see `BabelfishCompassUser.cfg `and `BabelfishCompassUser.Optimistic.cfg` under `C:\Users\[user name]\Documents\BabelfishCompass` now.
Thank you for your support. Much appreciated!
